src/app/loadout-drawer/loadout-drawer-reducer.ts

```diff
@@ -467,49 +394,43 @@ export function fillLoadoutFromEquipped(
     const newEquippedItems = store.items.filter(
       (item) => item.equipped && itemCanBeInLoadout(item) && itemMatchesCategory(item, category)
     );
-    const mods: number[] = [];
+    const modsByBucket: { [bucketHash: number]: number[] } = {};
     for (const item of newEquippedItems) {
       if (!(item.bucket.hash in equippedItemsByBucket)) {
-        const loadoutItem = convertToLoadoutItem(item, true, 1);
-        if (item.bucket.hash === BucketHashes.Subclass) {
-          loadoutItem.socketOverrides = createSocketOverridesFromEquipped(item);
+        loadout = addItem(defs, item, true)(loadout);
+
+        // Only save fashion for the items we added
+        const plugs = item.sockets
+          ? filterMap(
+              getSocketsByCategoryHash(item.sockets, SocketCategoryHashes.ArmorCosmetics),
+              (s) => s.plugged?.plugDef.hash
+            )
+          : [];
+        if (plugs.length) {
+          modsByBucket[item.bucket.hash] = plugs;
         }
-        loadout.items.push(loadoutItem);
-        mods.push(...extractArmorModHashes(item).map(mapToNonReducedModCostVariant));
       }
     }
-    if (mods.length && (loadout.parameters?.mods ?? []).length === 0) {
-      loadout.parameters = {
-        ...loadout.parameters,
-        mods,
-      };
-    }
-    if (artifactUnlocks?.unlockedItemHashes.length) {
-      loadout.parameters = {
-        ...loadout.parameters,
-        artifactUnlocks,
-      };
+
+    // Populate mods if they aren't already there
+    if (!category && _.isEmpty(loadout.parameters?.mods)) {
+      loadout = syncModsFromEquipped(store)(loadout);
     }
-    // Save "fashion" mods for equipped items
-    const modsByBucket: { [bucketHash: number]: number[] } = {};
-    for (const item of newEquippedItems.filter((i) => i.bucket.inArmor)) {
-      const plugs = item.sockets
-        ? filterMap(
-            getSocketsByCategoryHash(item.sockets, SocketCategoryHashes.ArmorCosmetics),
-            (s) => s.plugged?.plugDef.hash
-          )
-        : [];
-      if (plugs.length) {
-        modsByBucket[item.bucket.hash] = plugs;
-      }
+
+    // Populate artifactUnlocks if they aren't already there
+    if (!category && _.isEmpty(loadout.parameters?.artifactUnlocks)) {
```

Comment:
This is a behavior change - previously artifact unlocks would always be overwritten whenever we used "fill in using equipped", whether it was the global or the per-category one. Now, only the global one fills them in, and only if they're not already saved.